src/kOS/Utilities/VesselUtils.cs

```diff
@@ -231,29 +231,64 @@ public static object TryGetEncounter(Vessel vessel, SharedObjects sharedObj)
             return "None";
         }
 
+               public static KSPActionParam makeActionParam(bool state) // just use this to when you need to call action that requires bool input for direction
+        {
+            if (state){ return new KSPActionParam(0, KSPActionType.Activate); }
+            else { return new KSPActionParam(0, KSPActionType.Deactivate);}
+        }
+
         public static void LandingLegsCtrl(Vessel vessel, bool state)
         {
-            // This appears to work on all legs in 0.22
-            vessel.rootPart.SendEvent(state ? "LowerLeg" : "RaiseLeg");
+            vessel.rootPart.SendEvent(state ? "LowerLeg" : "RaiseLeg"); //legacy
+
+            foreach (var p in vessel.parts)
+            {
+                var geardeploy = p.FindModulesImplementing<ModuleWheels.ModuleWheelDeployment>();
+                foreach (var gd in geardeploy)
+                {
+                    var gear = p.Modules[gd.baseModuleIndex] as ModuleWheelBase;
+                    if ((gear != null) && (gear.wheelType == WheelType.LEG)) // identify leg
+                    {gd.ActionToggle(makeActionParam(state)); }
+                }
+            }
         }
 
-        internal static object GetLandingLegStatus(Vessel vessel)
+        public static object GetLandingLegStatus(Vessel vessel)
         {
             var atLeastOneLeg = false; // No legs at all? Always return false
 
             foreach (var p in vessel.parts)
             {
-                if (!p.Modules.OfType<ModuleLandingLeg>().Any()) continue;
-                atLeastOneLeg = true;
+                if (p.Modules.OfType<ModuleLandingLeg>().Any()) //Legacy
```

Comment:
Maybe we don't need that legacy code anymore. By this point most mods would likely have this updated to the new system. The stock had fully switched right away, including automatic conversion on existing crafts.